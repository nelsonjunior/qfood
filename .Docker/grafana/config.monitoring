GF_SECURITY_ADMIN_PASSWORD=qfood
GF_USERS_ALLOW_SIGN_UP=false