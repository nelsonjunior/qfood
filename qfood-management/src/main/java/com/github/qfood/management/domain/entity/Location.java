package com.github.qfood.management.domain.entity;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;

import javax.persistence.*;

@Entity
@Table(name = "location")
public class Location extends PanacheEntityBase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    public Double latitude;

    public Double longitude;
}
