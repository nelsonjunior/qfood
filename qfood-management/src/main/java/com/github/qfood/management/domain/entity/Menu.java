package com.github.qfood.management.domain.entity;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;

import javax.persistence.*;
import java.math.BigDecimal;

@Entity
@Table(name = "menu")
public class Menu extends PanacheEntityBase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    public String name;
    public String description;
    public BigDecimal price;

    @ManyToOne
    public Restaurant restaurant;

}
