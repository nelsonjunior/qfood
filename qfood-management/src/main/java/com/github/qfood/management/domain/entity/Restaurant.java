package com.github.qfood.management.domain.entity;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name = "restaurant")
public class Restaurant extends PanacheEntityBase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    public String owner;
    public String documentID;
    public String name;

    @ManyToOne
    public Location location;

    @CreationTimestamp
    public Date dateCreatedAt;

    @UpdateTimestamp
    public Date dataUpdatedAt;
}
