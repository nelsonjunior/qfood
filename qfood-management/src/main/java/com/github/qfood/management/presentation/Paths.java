package com.github.qfood.management.presentation;

public final class Paths {

    public static final String RESTAURANTS = "/restaurants";
}
