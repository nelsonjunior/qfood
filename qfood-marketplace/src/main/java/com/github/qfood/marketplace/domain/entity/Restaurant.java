package com.github.qfood.marketplace.domain.entity;

import java.util.Date;

public class Restaurant {

    public Long id;

    public String owner;
    public String documentID;
    public String name;

    public Location location;

    public Date dateCreatedAt;

    public Date dataUpdatedAt;

    @Override
    public String toString() {
        return "Restaurant{" +
                "id=" + id +
                ", owner='" + owner + '\'' +
                ", documentID='" + documentID + '\'' +
                ", name='" + name + '\'' +
                ", location=" + location +
                ", dateCreatedAt=" + dateCreatedAt +
                ", dataUpdatedAt=" + dataUpdatedAt +
                '}';
    }
}
