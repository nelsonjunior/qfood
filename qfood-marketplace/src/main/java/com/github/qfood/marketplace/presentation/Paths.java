package com.github.qfood.marketplace.presentation;

public final class Paths {

    public static final String RESTAURANTS = "/restaurants";
    public static final String MENUS = "/menus";
    public static final String CARTS = "/carts";
}
