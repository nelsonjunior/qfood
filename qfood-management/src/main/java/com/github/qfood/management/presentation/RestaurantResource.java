package com.github.qfood.management.presentation;

import com.github.qfood.management.domain.entity.Menu;
import com.github.qfood.management.domain.entity.Restaurant;
import com.github.qfood.management.exception.ServiceException;
import com.github.qfood.management.repository.RestaurantRepository;
import com.github.qfood.management.service.MenuService;
import org.eclipse.microprofile.openapi.annotations.tags.Tag;
import org.eclipse.microprofile.openapi.annotations.tags.Tags;
import org.jboss.logging.Logger;

import javax.inject.Inject;
import javax.transaction.Transactional;
import javax.ws.rs.*;
import javax.ws.rs.core.*;
import java.util.List;
import java.util.Optional;
import java.util.stream.Collectors;

@Path(Paths.RESTAURANTS)
@Produces(MediaType.APPLICATION_JSON)
@Consumes(MediaType.APPLICATION_JSON)
@Tag(name = "restaurant")
public class RestaurantResource {

    private static final Logger LOGGER = Logger.getLogger(RestaurantResource.class);

    @Inject
    RestaurantRepository restaurantRepository;

    @Inject
    MenuService menuService;

    @GET
    public Response getAllRestaurants() {
        List<Restaurant> restaurants = restaurantRepository.listAll();
        LOGGER.debug("Total number of restaurants " + restaurants);
        return Response.ok(restaurants).build();
    }

    @GET
    @Path("/{id}")
    public Response getRestaurant(@PathParam("id") Long id) {
        Optional<Restaurant> restaurant = restaurantRepository.findByIdOptional(id);
        if (restaurant != null) {
            LOGGER.debug("Found restaurant " + restaurant);
            return Response.ok(restaurant).build();
        } else {
            LOGGER.debug("No hero found with id " + id);
            return Response.noContent().build();
        }
    }

    @POST
    @Transactional
    public Response add(Restaurant dto, @Context UriInfo uriInfo) {
        dto.persist();
        UriBuilder builder = uriInfo.getAbsolutePathBuilder().path(Long.toString(dto.id));
        LOGGER.debug("New restaurant created with URI " + builder.build().toString());
        return Response.created(builder.build()).build();
    }

    @PUT
    @Path("{id}")
    @Transactional
    public void update(@PathParam("id") Long id, Restaurant dto) {
        Optional<Restaurant> restaurantEntity = Restaurant.findByIdOptional(id);
        if (restaurantEntity.isEmpty()) {
            throw new NotFoundException();
        }
        Restaurant restaurant = restaurantEntity.get();
        restaurant.name = dto.name;
        restaurant.persist();
    }

    @DELETE
    @Path("{id}")
    @Transactional
    public void delete(@PathParam("id") Long id) {
        Optional<Restaurant> restaurantEntity = Restaurant.findByIdOptional(id);
        restaurantEntity.ifPresentOrElse(Restaurant::delete, () -> {
            throw new NotFoundException();
        });
    }

    @GET
    @Path("/{idRestaurant}/menus")
    @Tag(name = "menu")
    public List<Menu> indexMenu(@PathParam("idRestaurant") Long idRestaurant) {
        try {
            return menuService.getMenusByRestaurantId(idRestaurant).collect(Collectors.toList());
        } catch (ServiceException e) {
            throw new NotFoundException(e.getMessage());
        }
    }

    @POST
    @Path("{idRestaurant}/menus")
    @Transactional
    @Tag(name = "menu")
    public Response addMenu(@PathParam("idRestaurant") Long idRestaurant, Menu dto) {
        try {
            menuService.insert(idRestaurant, dto);
            return Response.status(Response.Status.CREATED).build();
        } catch (ServiceException e) {
            throw new NotFoundException(e.getMessage());
        }
    }

    @PUT
    @Path("{idRestaurant}/menus/{id}")
    @Transactional
    @Tag(name = "menu")
    public Response updateMenu(@PathParam("idRestaurant") Long idRestaurant, @PathParam("id") Long id, Menu dto) {
        try {
            menuService.update(idRestaurant, id, dto);
            return Response.status(Response.Status.CREATED).build();
        } catch (ServiceException e) {
            throw new NotFoundException(e.getMessage());
        }
    }

    @DELETE
    @Path("{idRestaurant}/menus/{id}")
    @Transactional
    @Tag(name = "menu")
    public void delete(@PathParam("idRestaurant") Long idRestaurant, @PathParam("id") Long id) {
        try {
            menuService.delete(idRestaurant, id);
        } catch (ServiceException e) {
            throw new NotFoundException(e.getMessage());
        }
    }
}
