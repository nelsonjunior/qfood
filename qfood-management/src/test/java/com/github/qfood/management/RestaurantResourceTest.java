package com.github.qfood.management;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import com.github.qfood.management.config.CadastroTestLifecycleManager;
import com.github.qfood.management.domain.entity.Restaurant;
import com.github.qfood.management.presentation.Paths;
import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import javax.ws.rs.core.Response.Status;

import static io.restassured.RestAssured.given;
import static javax.ws.rs.core.HttpHeaders.ACCEPT;
import static javax.ws.rs.core.HttpHeaders.CONTENT_TYPE;
import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertTrue;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(CadastroTestLifecycleManager.class)
public class RestaurantResourceTest {

    @Test
    @DataSet("restaurants-usecase-1.yml")
    public void testFindAllRestaurants() {
        String result = given()
                .when().get(Paths.RESTAURANTS)
                .then()
                .statusCode(200)
                .extract()
                .asString();
        Approvals.verifyJson(result);
    }

    @Test
    public void testInsertRestaurant() {
        Restaurant dto = new Restaurant();
        dto.name = "Japonese Restaurant";
        dto.owner = "ID owner";
        String location = given()
                .body(dto)
                .header(CONTENT_TYPE, APPLICATION_JSON)
                .header(ACCEPT, APPLICATION_JSON)
                .when().post(Paths.RESTAURANTS)
                .then()
                .statusCode(Status.CREATED.getStatusCode())
                .extract().header("Location");

        assertTrue(location.contains(Paths.RESTAURANTS));

        // Stores the id
        String[] segments = location.split("/");
        String restaurantId = segments[segments.length - 1];
        assertNotNull(restaurantId);

        String result = given()
                .pathParam("id", restaurantId)
                .when().get(Paths.RESTAURANTS + "/{id}")
                .then()
                .statusCode(200)
                .extract()
                .asString();
        assertNotNull(result);

    }
}
