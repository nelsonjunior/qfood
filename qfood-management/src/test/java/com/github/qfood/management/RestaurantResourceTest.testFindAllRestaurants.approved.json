[
  {
    "id": 123,
    "owner": "ID keyclock",
    "name": "Brazilian Restaurant"
  },
  {
    "id": 234,
    "owner": "ID keyclock 2",
    "name": "Mexican Restaurant"
  }
]